coalahtml/tree.py

```diff
@@ -0,0 +1,81 @@
+from collections import OrderedDict
+from queue import Queue
+import os
+
+
+class Node:
+    """
+    Node class that defines an individual node.
+    """
+
+    def __init__(self, path):
+        self.child = {}
+        self.parent = None
+        self.path = path
+        self.abspath = ""
+        self._id = id(self)
+
+
+class Tree:
+
+    def __init__(self, file_list):
+        """
+        Create a root and build the graph.
+        """
+        self.root = Node(os.path.commonprefix(file_list))
```

Comment:
hmm so `commonprefix` belongs to `path` but it looks for string prefixes. I am disappointed with `commonprefix` method here. :disappointed: